src/layout/actions/layout.rs

```diff
@@ -901,18 +924,41 @@ impl LayoutTree {
             let parent_ix = self.tree.parent_of(child_ix)
                 .expect("Node had no parent");
             let children = self.tree.children_of(parent_ix);
-            let index = children.iter().position(|&node_ix| node_ix == child_ix)
-                .map(|num| (num + 1).to_string());
+            let index = children.iter().position(|&node_ix| node_ix == child_ix);
+            let index_str = index.map(|num| (num + 1).to_string());
+
+            // I can't do it inside the match ahead because it borrows
+            // self.tree, and I need self.tree to get the children's titles
+            let titles = children.iter().map(|&node_ix| {
+                match self.tree[node_ix] {
+                    Container::Container { ref mut borders, .. } =>
+                        borders.as_ref().map( |b| b.title() )
+                            .unwrap_or("").into(),
+                    Container::View { ref mut borders, .. } =>
+                        borders.as_ref().map( |b| b.title() )
+                            .unwrap_or("").into(),
+                    ref child => {
+                        error!("Container child is {:#?}", child);
+                        panic!("Container childr is not a Container nor a View")
+                    }
+                }
+            }).collect();
+
             if !self.tree.on_path(child_ix) && Some(child_ix) != self.active_container {
                 self.set_borders(child_ix, borders::Mode::Inactive)?;
             } else {
                 match self.tree[parent_ix] {
                     Container::Container { layout, ref mut borders, .. } => {
                         if layout == Layout::Tabbed || layout == Layout::Stacked {
                             borders.as_mut().map(|b| {
+
+                                b.set_children(titles, index.unwrap());
```

Comment:
Though unlikely, I don't like this `unwrap` here. Can you instead make this a `TreeError` and return early instead?

At that point you won't loop through the rest if there's an error, but it's better than unwrapping.

Reply:
What does it mean when index is None?

Reply:
When the `index` is `None`, that means the object couldn't be found in the list (e.g the node wasn't a child of the parent). Now, technically this is impossible with the code we have now. This is a thing that should never happen (tm). So an unwrap is _ok_, but even for the "impossible" errors I think it's best we handle them gracefully.

If that seems like too much work, doing `expect("The node index was not a child!")` would be better because it at least gives a reason why this occured.